sim/src/image.rs

```diff
@@ -1294,17 +1296,26 @@ fn install_image(flash: &mut SimMultiFlash, slot: &SlotInfo, len: usize,
     tlv.add_bytes(&b_img);
 
     // Generate encrypted images
-    let flag = TlvFlags::ENCRYPTED as u32;
-    let is_encrypted = (tlv.get_flags() & flag) == flag;
+    let flag = TlvFlags::ENCRYPTED_AES128 as u32 | TlvFlags::ENCRYPTED_AES256 as u32;
+    let is_encrypted = ((tlv.get_flags() & flag) == TlvFlags::ENCRYPTED_AES128 as u32)
+                    || ((tlv.get_flags() & flag) == TlvFlags::ENCRYPTED_AES256 as u32);
```

Comment:
A nit since the code is already there, but this would be easier to do by just `(tlv.get_flags() & flag) != 0`, which would work for any number of flags, or a single one as well...